src/libraries/System.Net.NameResolution/tests/FunctionalTests/GetHostEntryTest.cs

```diff
@@ -256,7 +274,9 @@ public async Task DnsGetHostEntry_LocalHost_ReturnsFqdnAndLoopbackIPs(int mode)
         [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
         [InlineData(0)]
         [InlineData(1)]
+#if !TARGET_WASI
         [InlineData(2)]
+#endif
```

Comment:
instead of doing it like this you can also do

```csharp
if (OperatingSystem.IsWasi() && mode == 2)
  throw new SkipTestException("mode 2 is not supported on WASI");
```

so you don't need the ifdef